test/app/dashboard.test.tsx

```diff
@@ -3,211 +3,256 @@ import {expect, describe, it} from 'vitest';
+        const submitButton = await screen.findByTestId(
+          'new-budget-form-button-submit'
+        );
+
+        await user.type(descriptionInput, "Dance till you can't any more");
+        await user.click(submitButton);
+        await user.click(submitButton);
+
+        const budgets = await mockDataprovider.getBudgets([], 25, defaultSort);
+        expect(budgets.length).toBe(0);
+      });
+
+      it('make budget form populates budget with correct data', async () => {
+        // setup the userEvents library
+        const user = userEvent.setup();
+
+        mockDatabase.emptyCollection(Collections.Budgets);
+        //render the daashboard
+        // await Dashboard since it is an async component
+        render(await Dashboard({...props}));
+
+        // await Dashboard since it is an async component
+
+        expect(
+          screen.queryByTestId('new-budget-form-button-add')
+        ).toBeInTheDocument();
+        // Click the new button form
+        await user.click(screen.getByTestId('new-budget-form-button-add'));
+
+        // submit a new budget
+        const nameInput = await screen.findByTestId('new-budget-form-input-name');
+        const descriptionInput = await screen.findByTestId(
+          'new-budget-form-input-description'
+        );
+        const dateInput = await screen.findByTestId(
+          'new-budget-form-input-datepicker'
+        );
+        const locationInput = await screen.findByTestId(
+          'new-budget-form-input-location'
+        );
+
+        const submitButton = await screen.findByTestId(
+          'new-budget-form-button-submit'
+        );
+
+        const name = 'Dance Party';
+        const description = "Dance till you can't any more";
+        const location = 'Main Pit';
+
+        await user.type(nameInput, name);
+        await user.type(descriptionInput, description);
+        await user.type(locationInput, location);
+
+        // Set date to the day after today
+        const today = new Date();
+        await user.click(dateInput);
+        await user.click(await screen.findByText(today.getDate()));
+
+        await user.click(submitButton);
+
+        const budgets = await mockDataprovider.getBudgets();
+
+        expect(budgets.length).toBe(1);
+        expect(budgets[0].event_name).toBe(name);
+        expect(budgets[0].event_description).toBe(description);
+        expect(budgets[0].event_location).toBe(location);
+        expect(new Date(budgets[0].event_datetime!).getDate()).toBe(
+          today.getDate()
+        );
+      });
+    }
   });
+}
+
+testDashboard({
+  id: 'test_user',
+  name: 'Test Org',
+  user_type: 'RSO',
+  remaining_budget: 100,
+  total_budget: 1000,
+  pending_event: 3,
+  planned_event: 5,
+  completed_event: 2,
+});
+testDashboard({
+  id: 'test_sga_user',
+  name: 'SGA Treasurer',
+  user_type: 'SGA_Treasurer',
+  remaining_budget: 100,
+  total_budget: 1000,
+  pending_event: 3,
+  planned_event: 5,
+  completed_event: 2,
 });
```

Comment:
i don't think parametrizing `describe` is valid due to the way `vitest` works. on top of this, I think the sga dashboard and regular dashboard tests should be dne separately